src/python/m5/util/convert.py

```diff
@@ -258,6 +264,18 @@ def toMemoryBandwidth(value):
 
 
 def toMemorySize(value):
+    if (
+        type(value) is str
+        and len(value) > 1
+        and value[-2] in binary_prefixes.keys()
+        and not "i" in value
+    ):
+        from m5.util import warn
+
+        warn(
+            f"Base 10 memory/cache size {value} will be cast to base 2"
+            + f" size {value[0:-2]}{value[-2].upper()}i{value[-1]}."
+        )
```

Comment:
I'm not going to blog it on this but i think you could save yourself a lot of effort here by just having a prefix lookup in "src/python/m5/utility/convert.py". Something like:

```py
si_base_10_to_2 = {
    "kB" : "KiB",
    "MB" : "MiB",
     #...  etc.
}
```

then just:

```py
if get_prefix(value) in si_base_10_to_2:
   correct_prefix = si_base_10_to_2
   # ... etc.
```

I also think a function for extractin the units from the string would be handy as demoed in `get_prefix` above.